Added README for renv introduction (#1)
WFUSM-BDS/guideline-reproducibility: README.Rmd

---
title: "guideline-reproducibility"
output: github_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library(palmerpenguins)

```

> The goal of guideline-reproducibility is to introduce a method for package management that is necessary for reproducibility when collaborating in R. 

Reviewer: ```suggestion
The goal of guideline-reproducibility is to introduce a method for package management that is helpful for reproducibility when collaborating in R. 
```

<BR>

## Motivating Example

> Perhaps the most frustrating part of collaboration in R is due to differences in installed packages in collaborators' system libraries. It is not uncommon for code that runs without error in your R session, to encounter an unusual error on a collaborator's machine due to package version differences. This can result in time-consuming debugging that takes time away from more important tasks.

Reviewer: ```suggestion
Collaboration in R may become frustrating if differences in installed packages in collaborators' system libraries cause code to fail. It is not uncommon for code that runs without error in your R session, to encounter an error on a collaborator's machine due to package version differences. This can result in time-consuming debugging that takes time away from more important tasks.
```

<BR>

## renv 

> The `renv` package seeks to solve this issue by creating separate a project library for all packages that are installed and used in your project's code. `renv` stores metadata for each package in a lockfile that can be shared and will allow others to install the same package versions into their project library.  

Reviewer: ```suggestion
The `renv` package can solve this issue by creating separate a project library for all packages that are installed and used in your project's code. `renv` stores metadata for each package in a lockfile that can be shared and will allow others to install the same package versions into their project library. Since the library is project-specific, you won't have to worry about `renv` messing with your global library of packages. 
```

- withr      [3.0.1]
These packages will be installed into "./guideline-reproducibility/renv/library/R-4.2/x86_64-w64-mingw32".

Do you want to proceed? [Y/n]: Y

Installing packages --------------------------------------------------------
- Installing generics ...                       OK [linked from cache]
- Installing fansi ...                          OK [linked from cache]
- Installing utf8 ...                           OK [linked from cache]
- Installing pillar ...                         OK [linked from cache]
- Installing pkgconfig ...                      OK [linked from cache]
- Installing tibble ...                         OK [linked from cache]
- Installing withr ...                          OK [linked from cache]
- Installing tidyselect ...                     OK [linked from cache]
- Installing dplyr ...                          OK [linked from cache]
Successfully installed 9 packages in 0.18 seconds.

```

You may think the next step is recording the package and its dependencies in the lockfile. You do this through taking a `snapshot` of the project. 

```{r snapshot, eval = FALSE}
renv::snapshot()
```

```
- The lockfile is already up to date.
```

> However, `renv` tells us the lockfile is already up to date and does not update the lockfile. This is because we actually aren't using `dplyr` anywhere in our code yet. This is nice because just having a package installed in your project library does not mean that it will get recorded in the lockfile which would later prompt your collaborator to install the package even though it's not used in the project.

Reviewer: Nice! I did not know about this.

CRAN -----------------------------------------------------------------------
- colorspace     [* -> 2.1-1]
- farver         [* -> 2.1.2]
- gtable         [* -> 0.3.5]
- isoband        [* -> 0.2.7]
- labeling       [* -> 0.4.3]
- lattice        [* -> 0.20-45]
- MASS           [* -> 7.3-58.2]
- Matrix         [* -> 1.5-3]
- mgcv           [* -> 1.8-42]
- munsell        [* -> 0.5.1]
- nlme           [* -> 3.1-162]
- viridisLite    [* -> 0.4.2]

RSPM -----------------------------------------------------------------------
- ggplot2        [* -> 3.5.1]
- RColorBrewer   [* -> 1.1-3]
- scales         [* -> 1.3.0]

Do you want to proceed? [Y/n]: Y

- Lockfile written to "./guideline-reproducibility/renv.lock".
```

<BR>

### Sharing with Collaborators

When your collaborator opens your shared project files on their own computer for the first time with `renv`, `renv` will be automatically installed, and they will be prompted to run

```{r restore, eval = FALSE}
renv::restore()
```

The function will download and install all packages of the proper versions recorded in the lockfile into the project-specific library. 

<BR>

### Try it yourself

> Clone `guidelines-reproducibility` on your own computer and run `renv::restore()` to install the currently recorded packages into your project library. Then experiment with adding new packages to `packages.R` or write R scripts using new packages to this project. Explore how you can use `renv::status` and `renv::snapshot` to get the lockfile back up to date.

Reviewer: this is a great idea. I did this on my computer and there was 1 step before I could run `restore()`

![image](https://github.com/user-attachments/assets/a8b04593-8da1-4d1a-b09d-928d25545374)

Author: Thank you! I'll make a note about that in the text then that they will encounter this.
